Fix --config option to properly set values

heygrady opened the pull request:
#45

This fixes a regression that broke `--config foo.bar`.

Bonus:
- Small refactor of `options.ts` to facilitate testing
- ~Corrects the interface for `CliOptions`~
- 100% coverage (not full tests) for `options.ts`

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/swc-project/cli?pullRequest=48) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/swc-project/cli?pullRequest=48) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/swc-project/cli?pullRequest=48) it.</sub>

kdy1 commented:
Seems like there are some issues with tests

heygrady commented:
Fixed the issue. I reverted the changes I made to the `CliOptions` interface. I had noticed that those values _may_ be undefined in isolation, like for the test I was running. It seems like that isn't an issue in practice (or, maybe should be addressed separately). This should pass all of the integration tests now.

kdy1 reviewed (approved):
Thanks!
